### ISSUE TITLE
Re-export unscripted's Option, Flag, Positional, and Rest?

### ISSUE THREAD

**Issue by seaneagan** [google/grinder.dart#287]
#286 will allow optionally annotating task parameters with them.  So it would be convenient to re-export those from `package:grinder/grinder.dart`, otherwise the user will need to add a dependency on `unscripted` to their pubspec (with a version that is compatible with grinder's unscripted dependency), and then import `package:unscripted/unscripted.dart` into their `grind/tool.dart` script.

However, we would probably [need to set and maintain a very narrow constraint](https://codereview.chromium.org/775413002/) on `unscripted`.  Which means that anyone who wanted to use unscripted for other scripts in their package would have a limited selection of versions, and potentially a difficult time of determining what that selection even is.

**Comment by devoncarew** [comment]
Yeah, I have concerns with specifying narrow version constraints. You can get into pathological issues with the constraint solver, including being unable to find any versions that satisfy the deps. Not sure what the best solution is.
